rdflib/plugins/sparql/evaluate.py

```diff
@@ -302,9 +302,24 @@ def evalPart(ctx: QueryContext, part: CompValue):
 
 def evalServiceQuery(ctx: QueryContext, part):
     res = {}
+    service_string = part.get("service_string", "")
+    match = re.match(
+        "^service (.*):(.*) [ \n]*{(.*)}[ \n]*$",
+        service_string,
+        re.DOTALL | re.I,
+    )
```

Comment:
Again, `term.part` already has the expanded URI so it may not be necessary to do an additional regex match to figure it out. I'm just not familiar with the query processor enough to know if `term.part` _always and reliably_ has the expanded URI.

Reply:
i am not very sure of what `term.part` is. But i will make further changes and add tests. So, this pr is still active. I just want to break first PR asap. So, that i will not hesitate contribute to it more. thanks for mentioning this issue.

Reply:
I'm not either, which is why I was hoping someone who was more familiar with the query processor would know how to fix this the best way.